lib/active_fedora/fixity_service.rb

```diff
@@ -2,26 +2,57 @@ module ActiveFedora
   class FixityService
     extend ActiveSupport::Concern
 
-    attr_accessor :target, :response
+    attr_accessor :target
 
-    # @param [String, RDF::URI] target url for a Fedora resource
+    # @param [String, RDF::URI] target url for a Fedora resource.
     def initialize(target)
       raise ArgumentError, 'You must provide a uri' unless target
       @target = target.to_s
     end
 
-    # Executes a fixity check on Fedora and saves the Faraday::Response.
+    def response
+      @response ||= fixity_response_from_fedora
+    end
+
+    # For backwards compat, check always insists on doing a new request.
+    # you might want verified? instead which uses a cached request.
     # @return true or false
     def check
-      @response = fixity_response_from_fedora
+      @response = nil
+      verified?
+    end
+
+    # Executes a fixity check on Fedora
+    # @return true or false
+    def verified?
       status.include?(success)
     end
 
+    # An array of 1 or more literals reported by Fedora.
+    # See 'success' for which one indicates fixity check is good.
     def status
       fixity_graph.query(predicate: premis_status_predicate).map(&:object) +
         fixity_graph.query(predicate: fedora_status_predicate).map(&:object)
     end
 
+    # the currently calculated checksum, as a string URI, like
+    # "urn:sha1:09a848b79f86f3a4f3f301b8baafde455d6f8e0e"
+    def expected_message_digest
+      fixity_graph.query(predicate: ::RDF::Vocab::PREMIS.hasMessageDigest).first.try(:object).try(:to_s)
```

Comment:
#to_base here gives me `""<urn:sha1:1a89571e25dd372563a10740a883e93f8af2d146>""`. I don't really want those angle brackets. I guess I could do that and strip the angle brackets off... but what's here now seems simpler to me. What do you think?

I don't really understand the right ways to interact with the RDF::Graph stuff so appreciate your advice. But I'd also rather not fight with it if what I have here now works fine, unless there are significant problems I don't know about with what I have now?